sandboxjs/fetch_sandbox.js

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+const branchInfo = require('./bamboo_plan_info.json');
+const path = require('path');
+var xml2js = require('xml2js');
+const fs = require('fs-extra');
+const fetch = require('node-fetch');
+const unzip = require('./unzip');
+
+// Get CDAP version from the pom file.
+async function getCDAPVersion() {
+  return new Promise((resolve, reject) => {
+    var parser = new xml2js.Parser();
+    fs.readFile(path.join(__dirname , '..', 'pom.xml'), function(err, data) {
+      if (err) {
+        reject(err);
+      }
+      parser.parseString(data, function (err, result) {
+        if (err) {
```

Comment:
would `err` be ambiguous here from the parent readfile callback?

Reply:
Could be; changed it to `parseErr`.